### DISCUSSION TITLE
Feature Request: Add a Star Button to Each Lesson in the Curriculum to indicate Special Lessons

### DISCUSSION THREAD

**Discussion by bharadr** [TheOdinProject/theodinproject#4767] [Ideas]
### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] The title of this issue follows the `Feature Request: brief description of feature request` format, e.g. `Feature Request: Add a dark mode to the website`
- [X] Would you like to work on this issue?

### Description of the Feature Request

When I go through a lesson in TOP, there is a button to indicate the lesson is completed. I want a second button (right next to the green completion button) that will allow me to highlight (via a yellow star) lessons that are special to me, and ones I'll likely come back and go over again in the future.

Here is what I am envisioning:

Unclicked:
<img width="815" alt="unstarred" src="https://github.com/TheOdinProject/theodinproject/assets/46429739/0a947657-a406-4327-bf94-ec644f18542e">

Clicked:
<img width="815" alt="starred" src="https://github.com/TheOdinProject/theodinproject/assets/46429739/dac21ef6-5106-4520-99d2-48f3134448f1">

Some lessons are harder to master than others, and even though I may complete a lesson, I make a note of going back to this lesson to review again in the future (when I am working on projects).

Other lessons are excellent sources of information, and to me reveal some very important insights. I want to highlight these lessons as well, to indicate that I learned something very special (to me) in this lesson.

A second button (with an empty Star icon when un-clicked, a yellow star when clicked) would allow me and all other users to highlight the lessons that were most important to them, whether they want to go back and review it, or whether they want to remind themselves this lesson covered important information.

### Acceptance criteria

- [] A Star Button is present on the lesson dashboard for each lesson
- [] The Star Button turns yellow whenever a user wants to highlight a certain lesson
- [] A tooltip exists for this button to explain its purpose when hovered over
- [] We explain the use of this button somewhere near the introduction for this course
- [] We preserve the Star Status for all lessons for each user. So when they log on / off, it is still saved.

**Comment by KevinMulhern** [comment]
Thanks for the suggestion @bharadr. This sounds a lot like something @ChargrilledChook's bookmarking feature should solve https://github.com/TheOdinProject/theodinproject/pull/4308
